tools/toolchains/mbed_toolchain.py

```diff
@@ -397,14 +397,27 @@ def get_arch_file(self, objects):
 
     # THIS METHOD IS BEING CALLED BY THE MBED ONLINE BUILD SYSTEM
     # ANY CHANGE OF PARAMETERS OR RETURN VALUES WILL BREAK COMPATIBILITY
-    def compile_sources(self, resources, inc_dirs=None):
+    def compile_sources(self, resources, inc_dirs=None, exclude_paths=None):
         # Web IDE progress bar for project build
         files_to_compile = (
             resources.get_file_refs(FileType.ASM_SRC) +
             resources.get_file_refs(FileType.C_SRC) +
             resources.get_file_refs(FileType.CPP_SRC)
         )
-        self.to_be_compiled = len(files_to_compile)
+        # Remove files from paths to be excluded from the build and create
+        # a compilation queue.
+        compile_queue = (
+            files_to_compile
+            if not exclude_paths
```

Comment:
I think the ternary operator here makes the code more difficult to read, a simple `if then else` would be clearer IMO.